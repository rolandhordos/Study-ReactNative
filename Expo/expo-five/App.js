// @flow

import React, { Component } from 'react'
import { View, Text, StyleSheet } from 'react-native'

// App is a top level component itself
export default class App extends Component {

  state = {
    count: 0,
  }

  render() {
    return (
      <View style={styles.container}>
          <Text style={styles.counter}>(count)</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  counter: {
    fontSize: 60,
    color: 'blue',
  }
})
