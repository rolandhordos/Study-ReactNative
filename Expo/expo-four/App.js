import React, { Component } from 'react'
import { View, Text, StyleSheet } from 'react-native'
import Carousel from '@rolandhordos/react-native-carousel'

export default class App extends Component {

  render() {
    return(
      <Carousel width={styles.slide.width} animate={false}>
        <View style={styles.slide}>
          <Text>Item 1</Text>
        </View>
        <View style={styles.slide}>
          <Text>Item 2</Text>
        </View>
        <View style={styles.slide}>
          <Text>Item 3</Text>
        </View>
      </Carousel>
    )
  }
}

const styles = StyleSheet.create({
  slide: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: 375,
    backgroundColor: '#00CED1', // Dark Turquoise
  }
})
