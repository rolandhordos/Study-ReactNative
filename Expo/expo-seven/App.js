// @flow

import React, { Component } from 'react'
import { View, Text } from 'react-native'

export default class Title extends Component<{}> {

  render() {
    return (<Text>TODO List</Text>)
  }

}
