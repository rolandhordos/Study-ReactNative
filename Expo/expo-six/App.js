// @flow

import React, { Component } from 'react'
import { View, Text, AppRegistry } from 'react-native'
import { createStore } from 'redux'

// Redux - Actions

const types = {
  INCREMENT: 'INCREMENT',
}

// Redux - Reducers

const reducer = (state, action) => {
  if (action.type === types.INCREMENT) {
    // return a complete representation of state
    return { count: state.count + 1 }
  }
  return state  // default course is a passthrough of input state
}

// Initialize the store
const initialState = {count: 0}
const store = createStore(reducer, initialState)

console.log('#Redux is ready')

// Throw some state changes at the store
store.dispatch({ type: types.INCREMENT })
store.dispatch({ type: types.INCREMENT })
store.dispatch({ type: types.INCREMENT })

export default class App extends Component<{}> {
  render() {
    return(
      <View style={{flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <Text style={{ fontSize: 100 }}>
        {store.getState().count}
      </Text>
      </View>
    )
  }
}
