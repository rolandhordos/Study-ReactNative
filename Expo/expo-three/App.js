import React from 'react';
import { View, Text, StyleSheet } from 'react-native';

export default class App extends React.Component {
  render() {
    return (
      <View style={styling.container}><Text>Hello World</Text></View>
    );
  }
}

const styling = StyleSheet.create({
    container: {
      flexDirection: 'column',  // layout vertically from the top
      flex: 0.5,                // use only 50% of the screen
      justifyContent: 'center', // vertical alignment
      alignItems: 'center',     // horizontal alignment
    }
});
