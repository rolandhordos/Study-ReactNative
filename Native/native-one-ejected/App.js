import React from 'react';
import { View, Text, ScrollView, StyleSheet } from 'react-native';

export default class App extends React.Component {
  render() {
    return (
      <View style={styles.launchScreen}>
        <View style={styles.topHalfContainer}>
          <Text>Hello World</Text>
        </View>
        <ScrollView style={styles.bottomHalfContainer}>
          <View style={styles.box}/>
          <View style={styles.box}/>
          <View style={styles.box}/>
          <View style={styles.box}/>
          <View style={styles.box}/>
        </ScrollView>
      </View>
    );
  }
}

// Establish some normalized standards of User Experience.  Additionally if we use standard property names we can
// aggregate from DRY models using Object.assign.

// All boxes should use the same corner radius
const corner = { borderRadius: 8 }
// Establish a consistent container for half-screen layouts
const halfScreenContainer = {
  flexDirection: 'column',  // layout vertically from the top
  flex: 1.0,                // use only 50% of the screen, which for 2 sibling views means each are 100%
}
// The top half of the screen will hold the centered Hello World greeting
let greetingContainer = {
  justifyContent: 'center', // vertical alignment
  alignItems: 'center',     // horizontal alignment
  backgroundColor: 'whitesmoke',
}
Object.assign(greetingContainer, halfScreenContainer)  // adopt the standard half-screen settings

// Assembly time.
const styles = StyleSheet.create({
  launchScreen: {
    flex: 1.0,
  },
  topHalfContainer: greetingContainer,
  bottomHalfContainer: halfScreenContainer,
  box: {
    backgroundColor: 'orange',
    width: '30%',   // cool I can scale the layout for larger screens
    height: 110,
    margin: 12,
    borderRadius: corner.borderRadius,
  }
});
