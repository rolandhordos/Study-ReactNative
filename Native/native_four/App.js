import React, { Component } from 'react'
import { View, Text, StyleSheet } from 'react-native'
import Carousel from '@rolandhordos/react-native-carousel'

export default class App extends Component {
  render() {
    return (
      <Carousel>
        <View style={styles.slide} width={styles.slide.width}>
          <Text>Slide 1</Text>
        </View>
        <View style={styles.slide} width={styles.slide.width}>
          <Text>Slide 2</Text>
        </View>
        <View style={styles.slide} width={styles.slide.width}>
          <Text>Slide 3</Text>
        </View>
      </Carousel>
    )
  }
}

const styles = StyleSheet.create({
  slide: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: 375,
  }
})
