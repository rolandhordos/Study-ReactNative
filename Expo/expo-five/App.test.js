// @flow

import React from 'react';
import App from './App';
import renderer from 'react-test-renderer'


it('renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toBeTruthy();
});

describe('App', ()=> {

  const app = renderer.create(<App/>)
  expect(app).toBeDefined()

  const appViewHierarchy = app.toJSON()
  expect(appViewHierarchy).toBeDefined()

  expect(appViewHierarchy).toMatchSnapshot()

  it('Has a Counter component', ()=> {

    const counterView = appViewHierarchy.children[0]
    expect(counterView.type).toBe('Text')
  })

  test('initial count is zero', ()=> {

    const appTree = app.toTree()
    expect(appTree.instance.state.count).toBe(0)
  })

})

describe('Exploring Jest', () => {

  test('test is valid within describe', () => {
    expect(true).toBeTruthy()
  })

  describe('Mocking', () => {

    // Generic problem to test mocking against.
    function forEach(items, callback) {
      for (var item of items) {
        callback(item)
      }
    }

    describe('allows verification of algorithms', () => {

      test('with a mock function', () => {
        const callbackMock = jest.fn()

        forEach(['one', 'two', 'three'], callbackMock)
        expect(callbackMock.mock.calls.length).toBe(3)
          // the first arg of the first call had value 'one'
        expect(callbackMock.mock.calls[0][0]).toBe('one')
        expect(callbackMock.mock.calls[2][0]).toBe('three')
      })

      describe('algorithms involving objects', () => {

        class Item {
          name: string = "is default"
          count: number = 0
          static maxCount: number = 2
          error: ?string

          increment = () => {
            this.count = this.count + 1
            this.validate()
            expect(this.name).toBe("is set")  // bound method test
          }
          validate = () => {
            if (this.count > Item.maxCount) {
              this.count = Item.maxCount
            }
          }
        }

        let item: Item

        beforeEach(() => {
          item = new Item()
        })

        test('mocked methods', () => {
          // Prepare mock structure - external method observation
          item.increment = jest.fn()
          // Run and Evaluate
          item.name = "is set"
          item.increment()
          item.increment()
          expect(item.increment.mock.calls.length).toBe(2)
        })

        test('provides state witnessed by the mock', () => {
          // Prepare mock structure
          item.validate = jest.fn()
          // Run and evaluate - we want to verify how the validation is performed
          item.name = "is set"
          item.increment()
          item.increment()
          expect(item.validate.mock.calls.length).toBeGreaterThanOrEqual(2) // validate may have been called more than twice
          // the count the validate mock saw should equal 2
          expect(item.validate.mock.instances[1].count).toBe(2)
        })

      })
    })  // algorithms involving objects

    it('provides truth beyond external observation', () => {
      class Item {
        static maxCount = 10
        _count: number = 0
        get count() {
          if (this._count > Item.maxCount) {
            return Item.maxCount
          }
          else {
            return this._count
          }
        }
        increment = () => {
          this._count = this._count + 1
        }
      }

      let item = new Item()
      for (var i = 0; i < 100; i++) {
        item.increment()
      }

      expect(item.count).toBe(10)   // getter filters the internal raw count

      // Snoop on
      item.increment = jest.fn()
      item.increment()
      expect(item.increment.mock.instances[0]._count).toBe(100) // surfaces inner state witnessed by the call
    })

  })  // describe mocking
})  // describe Exploring Jest

describe('Exploring JS', () => {

  describe('Classes', () => {

    it('Allows an explicit constructor', () => {

      class Item {
        name: ?string
        constructor(name) {
          this.name = name
        }
      }

      const item = new Item('one')
      expect(item.name).toBe('one')
    })

    it('Allows for default property values', () => {

      class Item {
        name: string = 'not yet named'
      }

      const item = new Item()
      expect(item.name).toBe('not yet named')
    })

  }) // describe Classes

  it('has fast enumeration', () => {
    const lots = [{ name: 'one'}, { name: 'two' }, { name: 'three'}]
    for (var each of lots) {
      console.log('saw ' + each.name)
    }
  })

  it('maintains "this" scope in method calls', () => {

    class Item {
      name : string = "is the default"
      count : number = 0
      incrementBy = (n: number) => {
        this.count = this.count + n
        expect(this.count).toBeGreaterThan(0)
        expect(this.name).toBe("is set")
      }
    }

    let item = new Item()
    item.name = "is set"
    item.incrementBy(2)
    item.incrementBy(0)
    expect(item.count).toBe(2)
  })
}) // describe Exploring JS
