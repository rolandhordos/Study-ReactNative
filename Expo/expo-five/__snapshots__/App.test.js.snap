// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "blue",
        "fontSize": 60,
      }
    }
  >
    (count)
  </Text>
</View>
`;
